spyder/api/widgets/toolbars.py

```diff
@@ -210,6 +210,16 @@ def add_item(self, action_or_widget: ToolbarItem,
                     self.add_item(item, section=section, before=before,
                                   before_section=before_section)
 
+    def remove_item(self, item_id: str):
+        """Remove action or widget from toolbar by id."""
+        item = self._item_map.pop(item_id)
+        for section in list(self._section_items.keys()):
+            section_items = self._section_items[section]
+            if item in section_items:
+                section_items.remove(item)
+            if len(section_items) == 0:
+                self._section_items.pop(section)
```

Comment:
It seems this new method only removes toolbar items from the maps saved in each toolbar, but doesn't hide or remove the actual action added to the toolbar.

Could you check that? Perhaps you need to call `_render` again.